### DISCUSSION TITLE
Progress callback

### DISCUSSION THREAD

**Discussion by pippo111** [101arrowz/fflate#24] [Q&A] [answered]
Hi @101arrowz!

You mentioned here https://github.com/101arrowz/fflate/issues/8#issuecomment-705696852 that progress callback should be pretty easy for zipping files. Is it implemented already or is it something still in feature request list ?

If there is some way how to do it, could you please provide some example ?

**Comment by 101arrowz** [comment, marked as answer]
A progress callback is simple with Gzip/Gunzip, Zlib/Unzlib, and Deflate/Inflate, but not with Zip/Unzip. This is a limitation of me trying to keep the code very simple. I'm soon planning to add support for incrementally adding files to a ZIP archive; would that be all you wanted, or do you also want detailed progress that describes the compression as it's happening on a per-file basis?

**Reply by pippo111** [reply]
Yes, incrementally adding files should do it. I imagine that we could easily create simple progress indicator based on that. That would be all I need.

**Reply by 101arrowz** [reply]
Alright, in a few days/a week I'll push an update with a new API for incrementally adding files.

**Reply by 101arrowz** [reply]
Merging into #26 - check there for an update.

**Comment by 101arrowz** [comment]
@pippo111, I've fixed this as of v0.5.0, check the new docs and ask if you have any concerns. Thanks for your interest and the feature request!

**Reply by pippo111** [reply]
Great, thank you. I will check it and let you know.

**Reply by 101arrowz** [reply]
Did this end up working for your application?

**Reply by pippo111** [reply]
Be patient please :) I will let you know this or next week. Your changes are definately something I will make use of.

**Reply by 101arrowz** [reply]
Sorry, I didn't mean to bother you, I was just wondering if you had any issues with it or had forgotten. Take your time, I'm not in a rush :)

**Reply by pippo111** [reply]
Ok, I've just tested it. It works great and satisfies my requirenments! Good to have this feature not only for progress callback, but overall for all stream operations. Thank you.
